cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSet.java

```diff
@@ -108,15 +112,32 @@
    * in the iterator returned in the PartitionConsumerResult.
    *
    * @param partitionConsumerState the state from which to start consuming from
-   * @param partitionFilter a filter which must match the partitions to be consumed
    * @param limit number of partitions, which once reached, will not add add more partitions committed by other
    *              transactions; the limit is checked after adding consuming all partitions of a transaction, so
    *              the total number of consumed partitions may be greater than this limit
+   * @param predicate a predicate which determines the partitions to be consumed
    * @return {@link PartitionConsumerResult} which holds the state of consumption as well as an iterator to the consumed
    * {@link Partition}s
    */
   PartitionConsumerResult consumePartitions(PartitionConsumerState partitionConsumerState,
```

Comment:
because this is an API, we can't just change this method. We need to keep the existing one and this as a new method. And deprecate the old one, or keep it for convenience.